rest_docs/rest-catalog-open-api-v0.1.yaml

```diff
@@ -0,0 +1,727 @@
+servers:
+  - url: http://127.0.0.1:1080
+    description: URL Used for Mock-Server Unit Tests
+# All routes are currently configured using an Authorization header.
+security:
+  - BearerAuth: []
+paths:
+  /v1/config:
+    get:
+      tags:
+        - Configuration API
+      summary: List all catalog configuration settings
+      operationId: getConfig
+      description:
+        All REST catalogs will be initialized by calling this route. This route
+        will return at least the minimum necessary metadata to initialize the
+        catalog. Optionally, it can also include server-side specific overrides.
+        For example, it might also include information used to initialize this catalog
+        such as the details of the Http connection pooling, etc. This route might
+        also advertise information about operations that are not implemented
+        so that the catalog can eagerly throw or go about another way of performing
+        the desired action.
+      responses:
+        default:
+          description: Server-Specific Configuration Overrides
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/IcebergConfiguration'
+        "400":
+          description: Unknown Error
+        "401":
+          description: Unauthorized
+
+  /v1/namespaces/{namespace}/tables/{table}:
+    parameters:
+      - name: namespace
+        in: path
+        description: Namespace the table is in
+        required: true
+        schema:
+          type: string
+        examples:
+          singlepart_namespace:
+            value: "accounting"
+          multipart_namespace:
+            value: "accounting.tax"
+      - name: table
+        in: path
+        description: Name of the table to load
+        required: true
+        schema:
+          type: string
+        example: "sales"
+    get:
+      tags:
+        - Catalog API
+      summary: Load a given table from a given namespace
+      operationId: loadTable
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/GetTableResponse'
+        '401':
+          description: Unauthorized
+        # Using 412, `Precondition Failed`, instead of 404, as 404 makes monitoring response codes from ELBs
+        #  very difficult - Hard to tell if clients or servers are misconfigured and calling non-existent routes
+        #  or missing routes versus expected error cases such as NoSuchTableException (expected meaning that
+        #  a person who is on call shouldn't be paged for this but 404 they might need to be).
+        '412':
+          description: NoSuchTableException
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/NoSuchTableError'
```

Comment:
I think that this is the error schema, but that `NoSuchTableException` would be thrown. Right, @kbendick?

Reply:
That's correct. Sorry I wrote a response to this but somehow it didn't go through.

This is the error schema (so the JSON wrapper for the response), but it is `NoSuchTableException`.

It would in theory look like the following, where `code` is application defined and can be looked up (open to removing that part but it has been useful at other organizations I've been in to have more specific error codes in the response):
```
{ "error":  { "message": "The given table does not exist", "type": "NoSuchTableException", "code": 41201 } }
```